python/pyarrow/_dataset.pyx

```diff
@@ -1547,6 +1549,9 @@ cdef class DirectoryPartitioning(Partitioning):
             Synonymous with infer_dictionary for backwards compatibility with
             1.0: setting this to -1 or None is equivalent to passing
             infer_dictionary=True.
+        schema : Schema, default None
+            Do not infer the schema, but confirm that partition values match
+            this schema and infer dictionary values as appropriate.
```

Comment:
```suggestion
            Use this schema instead of inferring a schema from partition values.
            Partition values will be validated against this schema before accumulation
            into the Partitioning's dictionary.
```